base/setup/usetup/mui.c

```diff
@@ -256,6 +256,189 @@ MUIGetString(
     return "<nostring>";
 }
 
+/**
+ * @MUIGetEntry
+ *
+ * Retrieves a MUI entry of a page, given the page number and the text ID.
+ *
+ * @param[in]   Page
+ *     The MUI (Multilingual User Interface) entry page number, as a unsigned long integer.
+ *
+ * @param[in]   TextID
+ *      The text identification number (ID), as a unsigned integer. The parameter is used to identify
+ *      its MUI properties like the coordinates, text style flag and its buffer content.
+ *
+ * @return
+ *     Returns a constant MUI entry.
+ *
+ */
+const MUI_ENTRY *
+MUIGetEntry(
+    IN ULONG Page,
+    IN INT TextID)
+{
+    const MUI_ENTRY * entry;
+    ULONG index;
+
+    /* Retrieve the entries of a MUI page */
+    entry = FindMUIEntriesOfPage(Page);
+    if (!entry)
+    {
+        DPRINT("MUIGetEntryData(): Failed to get the translated entry page!\n");
+        return NULL;
+    }
+
+    /* Loop over the ID entries and check if it matches with one of them */
+    for (index = 0; entry[index].Buffer != NULL; index++)
+    {
+        if (entry[index].TextID == TextID)
+        {
+            /* They match so return the MUI entry */
+            return &entry[index];
+        }
+    }
+
+    /* Page number or ID are incorrect so in this case bail out */
+    return NULL;
+}
+
+/**
+ * @MUIClearText
+ *
+ * Clears a portion of text from the console output.
+ *
+ * @param[in]   Page
+ *     The MUI (Multilingual User Interface) entry page number, as a unsigned long integer.
+ *
+ * @param[in]   TextID
+ *      The text identification number (ID), as an integer. The parameter is used to identify
+ *      its MUI properties like the coordinates, text style flag and its buffer content.
+ *
+ * @return
+ *     Nothing.
+ *
+ */
+VOID
+MUIClearText(
+    IN ULONG Page,
+    IN INT TextID)
+{
+    const MUI_ENTRY * entry;
+
+    /* Get the MUI entry */
+    entry = MUIGetEntry(Page, TextID);
```

Comment:
For safety measure, check whether the returned `entry` is NULL and if so, just return.